File: UdemyCoursePy/src/mx/irving/stacksandqueuesandmore/Stack.py
class Stack(object):

	def __init__(self):
		self.items = []
	
	def isEmpty(self):
		return self.items == []
		
	def push(self, item):
		self.items.append(item)
		
	def pop(self):
		return self.items.pop()
	
	def peek(self):
		return self.items[len(self.items) - 1]
	
	def size(self):
		return len(self.items)
		
stack = Stack()
print(stack.isEmpty())
stack.push(1)
stack.push('Cadena')
stack.push(True)
print(stack.isEmpty())
print(stack.size())
print(stack.pop())
print(stack.size())
print(stack.peek())
print(stack.size())
print(stack.pop())
print(stack.pop())
print(stack.isEmpty())

File: UdemyCourseKt/src/mx/irving/queuesandstacks/Stack.kt
package mx.irving.queuesandstacks

class Stack {
    var data = mutableListOf<Any>()

    fun isEmpty() = data.isEmpty()

    fun push(item: Any) {
        data.add(item)
    }

    fun pop(): Any? = if (data.isEmpty()) null else data.removeAt(data.size - 1)

    fun peek(): Any? = data.lastOrNull()

    fun size() = data.size
}

fun main() {
    val stack = Stack()
    println(stack.isEmpty())
    stack.push("One")
    println(stack.size())
    stack.push(2)
    println(stack.isEmpty())
    stack.push(true)
    println(stack.size())
    println(stack.pop())
    println(stack.size())
    println(stack.peek())
    println(stack.size())
}